nipype/utils/misc.py

```diff
@@ -229,3 +239,57 @@ def normalize_mc_params(params, source):
         params[-1:2:-1] = aff2euler(matrix)
 
     return params
+
+
+def dict_diff(dold, dnew, indent=0):
+    """Helper to log what actually changed from old to new values of
+    dictionaries.
+
+    typical use -- log difference for hashed_inputs
+    """
+    # First check inputs, since they usually are lists of tuples
+    # and dicts are required.
+    if isinstance(dnew, list):
+        dnew = dict(dnew)
+    if isinstance(dold, list):
+        dold = dict(dold)
+
+    # Compare against hashed_inputs
+    # Keys: should rarely differ
+    new_keys = set(dnew.keys())
+    old_keys = set(dold.keys())
+
+    diff = []
+    if new_keys - old_keys:
+        diff += ["  * keys not previously seen: %s" % (new_keys - old_keys)]
+
+    if old_keys - new_keys:
+        diff += ["  * keys not presently seen: %s" % (old_keys - new_keys)]
+
+    # Add topical message
+    if diff:
+        diff.insert(0, "Dictionaries had differing keys:")
+
+    diffkeys = len(diff)
+
+    # Values in common keys would differ quite often,
+    # so we need to join the messages together
+    for k in new_keys.intersection(old_keys):
+        same = False
```

Comment:
not needed

Reply:
you sure? how would you modify this?

Reply:
it's too late in Europe to be sure about anything, but regardless what happens in this loop, `same` should be defined/changed either in `try` part or `except` part, so looked to me that it doesn't have to be defined before `try/except`

Reply:
Added to #2320. Even though this PR is changing a lot already, I'd leave this catch for a new one.

Reply:
@djarecka I agree that you can remove this one line. `same` is defined before being accessed in all branches here.

(This comment in response to #2387.)